File: app/src/main/java/io/neoterm/view/eks/StatedControlButton.java
package io.neoterm.view.eks;

import android.view.View;
import android.widget.ToggleButton;

import io.neoterm.view.ExtraKeysView;

/**
 * @author kiva
 */

public class StatedControlButton extends ControlButton {
    public ToggleButton toggleButton;
    public boolean initState;

    public StatedControlButton(String text, boolean initState) {
        super(text);
        this.initState = initState;
    }

    public StatedControlButton(String text) {
        this(text, false);
    }

    @Override
    public void onClick(View view) {
        setStatus(toggleButton.isChecked());
    }

    public void setStatus(boolean status) {
        toggleButton.setChecked(status);
        toggleButton.setTextColor(status ? ExtraKeysView.SELECTED_TEXT_COLOR : ExtraKeysView.NORMAL_TEXT_COLOR);
    }

    public boolean readState() {
        if (toggleButton.isPressed()) return true;
        boolean result = toggleButton.isChecked();
        if (result) {
            toggleButton.setChecked(false);
            toggleButton.setTextColor(ExtraKeysView.NORMAL_TEXT_COLOR);
        }
        return result;
    }
}

File: app/src/main/java/io/neoterm/view/eks/StatedControlButton.kt
package io.neoterm.view.eks

import android.view.View
import android.widget.ToggleButton

/**
 * @author kiva
 */

open class StatedControlButton @JvmOverloads constructor(text: String, var initState: Boolean = false) : ControlButton(text) {
    var toggleButton: ToggleButton? = null

    override fun onClick(view: View) {
        setStatus(toggleButton?.isChecked)
    }

    fun setStatus(status: Boolean?) {
        val button = toggleButton
        if (button != null && status != null) {
            button.isChecked = status
            button.setTextColor(
                    if (status) ExtraButton.SELECTED_TEXT_COLOR
                    else ExtraButton.NORMAL_TEXT_COLOR
            )
        }
    }

    fun readState(): Boolean {
        val button = toggleButton ?: return false

        if (button.isPressed) return true
        val result = button.isChecked
        if (result) {
            button.isChecked = false
            button.setTextColor(ExtraButton.NORMAL_TEXT_COLOR)
        }
        return result
    }
}
